ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift

```diff
@@ -150,7 +151,27 @@ class NavigationControllerDemoController: DemoController {
 
 // MARK: - NavigationControllerDemoController: UIGestureRecognizerDelegate
 
-extension NavigationControllerDemoController: UIGestureRecognizerDelegate {
+extension NavigationControllerDemoController: UIGestureRecognizerDelegate, SearchBarDelegate {
```

Comment:
can we make sure there are ways for this demo to test non spinner case?

Reply:
as implemented in delegate, when user press search button, i.e. on searchBarDidRequestSearch invocation we are showing the animated progress bar.

---
In reply to: [465868161](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465868161) [](ancestors = 465868161)

Reply:
As I mentioned before, I think it'd be nice to have a toggle that you can set the spinner visibility.
Right now it's hard to discover the spinner since we have to type text and hit "Search" to see it.

Reply:
I find following issues in adding a switch on the nav bar buttons:
1. it doesn't get sized/positioned correctly as it tried implementing it. Also, it gets primary color which gets mixed up with the nav bar's color
2. when we tap on search button, search bar moves up, hiding this switch and hence at that time we don't have a switch.
Also, I think it is far from a real use when users will want a spinner.
Couple of options that I could think of other than the approach in CR are:
1. having a separate demo view controller similar which showcases search bar's. But then the only extra functionality that we are displaying there would be the spinner
2. Instead of putting the switch in navigation bar button items, we can add it as subview [label + switch]. But the addRow functionality is available in DemoController which isn't inherited by the controllers that we are showcasing here inside navigation controller's demo.
I would like to discuss it before making change.

---
In reply to: [468751778](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468751778) [](ancestors = 468751778)